src/capability.ts

```diff
@@ -0,0 +1,103 @@
+import * as ability from "./capability/ability"
+import * as resourcePointer from "./capability/resource-pointer"
+import * as superUser from "./capability/super-user"
+import * as util from "./util"
+
+import { Ability, isAbility } from "./capability/ability"
+import { ResourcePointer, isResourcePointer } from "./capability/resource-pointer"
+import { Superuser, SUPERUSER } from "./capability/super-user"
+
+
+// RE-EXPORTS
+
+
+export { ability, resourcePointer, superUser }
+
+
+
+// 💎
+
+
+export type Capability = {
+  with: ResourcePointer
+  can: Ability
+}
+
+export type EncodedCapability = {
+  with: string
+  can: string
+}
+
+
+
+// TYPE CHECKS
+
+
+export function isCapability(obj: unknown): obj is Capability {
+  return util.isRecord(obj)
+    && util.hasProp(obj, "with") && isResourcePointer(obj.with)
+    && util.hasProp(obj, "can") && isAbility(obj.can)
+}
+
+export function isEncodedCapability(obj: unknown): obj is EncodedCapability {
+  return util.isRecord(obj)
+    && util.hasProp(obj, "with") && typeof obj.with === "string"
+    && util.hasProp(obj, "can") && typeof obj.can === "string"
+}
+
+
+
+// 🌸
+
+
+export function as(identifier: string): Capability {
+  return {
+    with: resourcePointer.as(identifier),
+    can: SUPERUSER
+  }
+}
+
+
+export function my(resource: Superuser | string): Capability {
+  return {
+    with: resourcePointer.my(resource),
+    can: SUPERUSER
+  }
+}
+
+
+export function prf(selector: Superuser | number, ability: Ability): Capability {
+  return {
+    with: resourcePointer.prf(selector),
+    can: ability
+  }
+}
```

Comment:
Should we have functions for `wnfs` and `email` too? I realise they're not part of the spec, but could be helpful?